### PR TITLE
feat(slack-notifications): make slack enabled by default

### PR DESCRIPTION
**Pull request by scefali**
Since the backfill (https://github.com/getsentry/sentry/pull/37321) for slack is complete, we can default to Slack being enabled for everyone.

### CODE REVIEW

**Comment by 97varun** [comment]
@scefali I don't understand this fully but we might be making assumptions about the defaults in a couple of places (that I know of) in the UI.

https://github.com/getsentry/sentry/blob/829ed407c1507c11385a5ef0251e3e6c6fa93d85/static/app/views/settings/account/notifications/notificationSettingsByType.tsx#L217-L220

https://github.com/getsentry/sentry/blob/829ed407c1507c11385a5ef0251e3e6c6fa93d85/static/app/views/settings/account/notifications/utils.tsx#L328-L343

**Comment by scefali** [comment]
> @scefali I don't understand this fully but we might be making assumptions about the defaults in a couple of places (that I know of) in the UI.
> 
> https://github.com/getsentry/sentry/blob/829ed407c1507c11385a5ef0251e3e6c6fa93d85/static/app/views/settings/account/notifications/notificationSettingsByType.tsx#L217-L220
> 
> https://github.com/getsentry/sentry/blob/829ed407c1507c11385a5ef0251e3e6c6fa93d85/static/app/views/settings/account/notifications/utils.tsx#L328-L343

@97varun I don't really understand why we have that comment. The backend is going to return `slack: "always"` so we probably don't even need that defaulting logic on the front-end like that. I can remove the defaulting logic on the front-end later in a follow up PR.

File: static/app/views/settings/account/notifications/notificationSettingsByType.tsx [hunk 1]
```diff
@@ -214,10 +214,10 @@ class NotificationSettingsByType extends AsyncComponent<Props, State> {
     const {notificationType} = this.props;
     const {notificationSettings} = this.state;
 
-    // NOTE: Update this when we want to enable Slack by default.
+    // TODO: Backend should be in charge of providing defaults since it depends on the type
     const provider = !isEverythingDisabled(notificationType, notificationSettings)
       ? getCurrentProviders(notificationType, notificationSettings)
-      : ['email'];
+      : ['email', 'slack'];
```
Comment (scefali) [inline static/app/views/settings/account/notifications/notificationSettingsByType.tsx:220]: @97varun I added this in the PR

**Review by leeandher — APPROVED** [review]
Code-wise this looks good 🚀, just a few comments

File: src/sentry/notifications/defaults.py [hunk 2]
```diff
@@ -48,8 +36,9 @@
 }
 
 
+# email and slack are defaulted to being on
 NOTIFICATION_SETTING_DEFAULTS = {
     ExternalProviders.EMAIL: NOTIFICATION_SETTINGS_ALL_SOMETIMES,
-    ExternalProviders.SLACK: NOTIFICATION_SETTINGS_SLACK,
-    ExternalProviders.MSTEAMS: NOTIFICATION_SETTINGS_MSTEAMS,
+    ExternalProviders.SLACK: NOTIFICATION_SETTINGS_ALL_SOMETIMES,
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/notifications/defaults.py:42]: @leeandher Correct, this is true for new users at least. We did a backfill for old users so they keep their old settings.

File: src/sentry/notifications/helpers.py [hunk 3]
```diff
@@ -36,21 +32,13 @@
...
     Approval notifications always default to Slack being on.
     """
-    from sentry.models import User
-
-    if (
-        isinstance(recipient, User)
-        and features.has("users:notification-slack-automatic", recipient, actor=recipient)
-        and provider == ExternalProviders.SLACK
-    ):
-        return NOTIFICATION_SETTINGS_ALL_SOMETIMES[type]
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/notifications/helpers.py:53]: @leeandher Teams are a little different. We only use them for issue alerts to my knowledge. And every time we link a team, we actually set the settings to always on: 
https://github.com/getsentry/sentry/blob/master/src/sentry/integrations/slack/views/link_team.py#L160-L165

And there's actually no way to disable this in settings. The experience for team notifications definitely needs some improvement.

**Pull request merged**